paymentsheet/src/main/java/com/stripe/android/link/ui/LinkContent.kt

```diff
@@ -198,6 +186,42 @@ private fun Screens(
     }
 }
 
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+private fun LinkNavHost(
+    navController: NavHostController,
+    initialDestination: LinkScreen,
+    modifier: Modifier = Modifier,
+    builder: NavGraphBuilder.() -> Unit
+) {
+    // True if the IME (soft keyboard) is animating.
+    val isImeAnimating = WindowInsets.imeAnimationSource != WindowInsets.imeAnimationTarget
+
+    // Cache the current screen size so we can size `LinkLoadingScreen` to match, reducing
+    // screen size animation jumps.
+    var screenSize by remember { mutableStateOf<IntSize?>(null) }
+    ProvideLinkScreenSize(screenSize) {
+        NavHost(
+            modifier = modifier.onSizeChanged { screenSize = it },
+            navController = navController,
+            startDestination = initialDestination.route,
+            enterTransition = { LinkScreenTransition.targetContentEnter },
+            exitTransition = { LinkScreenTransition.initialContentExit },
+            // Workaround a race condition where the route changes while the soft keyboard is
+            // animating in/out. Imagine navigating from screen A to B, where both screens are
+            // supposed to be equal height. If the soft keyboard closes immediately after the
+            // screen transition starts, the target height of screen B, which is locked in at
+            // the start of the animation, will be too short.
+            sizeTransform = if (isImeAnimating) {
```

Comment:
:eyes: Demo of the problem:

https://github.com/user-attachments/assets/67102000-7ad2-4fcb-8fa6-64fcbbac6fb9